cmake/functional.cmake

```diff
@@ -14,14 +14,15 @@ function(GENERATE_FUNCTIONAL_API_AND_PYBIND11_CPP SRCS HDRS PYBIND_SRCS ROOT_DIR
   list(APPEND HDRS ${PROJECT_BINARY_DIR}/${GENERATED_API_DIR}/functional_api.yaml.h)
   list(APPEND PYBIND_SRCS ${PROJECT_BINARY_DIR}/${GENERATED_PYBIND_DIR}/functional_api.yaml.pybind.cpp)
 
-  add_custom_command(
-      OUTPUT "${PROJECT_BINARY_DIR}/${GENERATED_API_DIR}/functional_api.yaml.cpp"
-             "${PROJECT_BINARY_DIR}/${GENERATED_API_DIR}/functional_api.yaml.h"
-             "${PROJECT_BINARY_DIR}/${GENERATED_PYBIND_DIR}/functional_api.yaml.pybind.cpp"
+  add_custom_target(
+      ALL
       COMMAND ${Python_EXECUTABLE} ${PROJECT_SOURCE_DIR}/tools/generate_functional_api.py
               --yaml_file_path ${YAML_FILE} --generate_pybind
       DEPENDS ${Python_EXECUTABLE} create_functional_api_dir create_functional_pybind_dir
               ${PROJECT_SOURCE_DIR}/tools/generate_functional_api.py ${YAML_FILE}
+      BYPRODUCTS "${PROJECT_BINARY_DIR}/${GENERATED_API_DIR}/functional_api.yaml.cpp"
```

Comment:
> BYPRODUCTS
> New in version 3.2.
> 
> Specify the files the command is expected to produce but whose modification time may or may not be updated on subsequent builds. If a byproduct name is a relative path it will be interpreted relative to the build tree directory corresponding to the current source directory. Each byproduct file will be marked with the GENERATED source file property automatically.

这个旧点的 cmake 会挂吧